raft/server.cc

```diff
@@ -1888,4 +1889,27 @@ std::ostream& operator<<(std::ostream& os, const server_impl& s) {
     return os;
 }
 
+future<> server_impl::override_snapshot_thresholds() {
+    return utils::get_local_injector().inject(
+        "raft_server_set_snapshot_thresholds", [this](auto& handler) -> future<> {
+            const auto set_parm = [&handler](auto& target, const std::string_view name) -> void {
+                const auto from = handler.get(name);
+                if (from) {
+                    try {
+                        target = boost::lexical_cast<std::remove_reference_t<decltype(target)>>(*from);
+                        logger.info("Applied _config.{}={}", name, *from);
+                    } catch (const boost::bad_lexical_cast& e) {
+                        logger.info("Could not apply a snapshot threshold param: {}, value: {}, error: {}",
+                                    name, *from, e.what());
+                    }
+                }
+            };
+
+            set_parm(_config.snapshot_threshold, "snapshot_threshold");
+            set_parm(_config.snapshot_threshold_log_size, "snapshot_threshold_log_size");
+            set_parm(_config.snapshot_trailing, "snapshot_trailing");
+            set_parm(_config.snapshot_trailing_size, "snapshot_trailing_size");
+            co_return;
```

Comment:
I'm wondering of the `co_return` is even needed here, maybe it can be omitted?

Reply:
This is a lambda that's called by error_injection::inject, and it's type defined as following `the waiting_handler_fun = std::function<future<>(injection_handler&)>; `, so I can't do that. however we can change the return type of override_snapshot_thresholds() to `void` by waiting for the injection result in the function

Reply:
Yeah ... it looks that according to the https://en.cppreference.com/w/cpp/language/coroutines it should be possible to omit the `co_return` for void promises, but apparently it requires that the promise type will have the `return_void()` defined, which we might be missing for our promise/future types in the Seastar.

So for the time being it looks like we need to keep the `co_return`.

Reply:
We omit `co_return` in plenty of places:

https://github.com/scylladb/scylladb/blob/607abe96e86480bc8bbe1005b21764fe20be7dfa/raft/server.cc#L1205

Reply:
it looks like the issue here might be that we need either `co_return` or `co_await` to make the compiler consider it a coroutine? so if we don't have either just hinting the return type as `-> future<>` might not be enough (?)

Reply:
If the lambda does not sleep it should not be a co-routine. We can `return make_ready_future<>()` here. Even if a co-routine does not sleep it allocates co-routine frame IIRC.

Reply:
@gleb-cloudius  Could you elaborate, what frame does a corutine allocate?

Reply:
changed to `return make_ready_future<>() ` in the new version

Reply:
> @gleb-cloudius Could you elaborate, what frame does a corutine allocate?

Co-routines are not free. They allocate a co-routine frame to hold all the local variables when preempted. Local variables are not on a stack frame in the co-routine case.